docs/v8-new-performance-apis.md

```diff
@@ -0,0 +1,250 @@
+happens transparently, and from an SDK perspective, all you have to think about are spans.
+
+## The Span schema
+
+Previously, spans & transactions had a bunch of properties and methods to be used. Most of these have been removed in
+favor of a slimmer, more straightforward API, which is also aligned with OpenTelemetry Spans. You can refer to the table
+below to see which things used to exist, and how they can/should be mapped going forward:
+
+| Old name              | Replace with                                         |
+| --------------------- | ---------------------------------------------------- |
+| `traceId`             | `spanContext().traceId`                              |
+| `spanId`              | `spanContext().spanId`                               |
+| `parentSpanId`        | Same                                                 |
+| `status`              | TODO: new signature                                  |
+| `sampled`             | `spanContext().traceFlags`                           |
+| `startTimestamp`      | `startTime` - note that this has a different format! |
+| `tags`                | Use `attributes` instead, or set tags on the scope   |
+| `data`                | Use `attributes` instead.                            |
+| `transaction`         | Removed                                              |
+| `instrumenter`        | Removed                                              |
+| `finish()`            | `end()`                                              |
+| `end()`               | Same                                                 |
+| `setTag()`            | `setAttribute()`                                     |
+| `setData()`           | `setAttribute()`                                     |
+| `setStatus()`         | TODO: new signature                                  |
+| `setHttpStatus()`     | ???                                                  |
+| `setName()`           | TODO: `updateName()`                                 |
+| `startChild()`        | Call `Sentry.startSpan()` independently              |
+| `isSuccess()`         | ???                                                  |
+| `toTraceparent()`     | ???                                                  |
+| `toContext()`         | ???                                                  |
+| `updateWithContext()` | ???                                                  |
+| `getTraceContext()`   | ???                                                  |
+
+In addition, a transaction has this API:
+
+| Old name                    | Replace with                                          |
+| --------------------------- | ----------------------------------------------------- |
+| `name`                      | Same                                                  |
+| `trimEnd`                   | Removed                                               |
```

Comment:
What should we do about `trimEnd`? 🤔 should we keep this somehow, or rework how this API works? I think with only spans this maybe also makes less sense overall, but not sure...?

Reply:
trimEnd is just about idleTransactions. @mydea is there a way in the new performance API to inspect the child spans for a particular span and set its endTimestamp to the highest timestamp of the child spans? If so, we can drop this.

Reply:
I think as we keep a list of the child spans, we should be able to keep the highest end timestamp and use this for `idleSpan.end(highestTimestamp)` 🤔 So I'd say this should be possible!